localstack/services/awslambda/lambda_executors.py

```diff
@@ -876,87 +733,127 @@ def prepare_event(self, environment, event_body):
-                ')";'
-                "%s "
-                '%s start -ai "$CONTAINER_ID";'
-            ) % (
-                docker_cmd,
-                entrypoint,
-                debug_docker_java_port,
-                common_flags,
-                docker_image,
-                command,
-                cp_cmd,
-                docker_cmd,
+            DOCKER_CLIENT.copy_into_container(container_id, f"{lambda_cwd}/.", DOCKER_TASK_FOLDER)
+            return DOCKER_CLIENT.start_container(
+                container_id, interactive=True, attach=True, stdin=stdin
             )
         else:
-            mount_flag = ""
+            mount_volumes = None
             if lambda_cwd:
-                mount_flag = '-v "%s":%s' % (
-                    Util.get_host_path_for_path_in_docker(lambda_cwd),
-                    DOCKER_TASK_FOLDER,
-                )
-            cmd = ("%s run -i" " %s" " %s" " %s" " %s %s") % (  # code mount  # common flags
-                docker_cmd,
-                entrypoint,
-                mount_flag,
-                common_flags,
-                docker_image,
-                command,
+                mount_volumes = [
+                    (Util.get_host_path_for_path_in_docker(lambda_cwd), DOCKER_TASK_FOLDER)
+                ]
+            return DOCKER_CLIENT.run_container(
+                image_name=docker_image,
+                interactive=True,
+                entrypoint=entrypoint,
+                remove=True,
+                network=network,
+                env_vars=env_vars,
+                dns=dns,
+                additional_flags=additional_flags,
+                command=command,
+                mount_volumes=mount_volumes,
+                stdin=stdin,
             )
-        return cmd
 
 
 class LambdaExecutorLocal(LambdaExecutor):
-    def _execute(self, func_arn, func_details, event, context=None, version=None):
+    def _execute_in_custom_runtime(self, cmd: str, func_details=None) -> InvocationResult:
+        """
+        Generic run function for executing lambdas in custom runtimes.
+
+        :param cmd: the command to execute
+        :param func_details: function details
+        :return: the InvocationResult
+        """
+
+        kwargs = {"stdin": True, "inherit_env": True, "asynchronous": True}
+        process = run(cmd, stderr=subprocess.PIPE, outfile=subprocess.PIPE, **kwargs)
+        result, log_output = process.communicate()
+        try:
+            result = to_str(result).strip()
+        except Exception:
+            pass
+        log_output = to_str(log_output).strip()
+        return_code = process.returncode
+
+        # Note: The user's code may have been logging to stderr, in which case the logs
+        # will be part of the "result" variable here. Hence, make sure that we extract
+        # only the *last* line of "result" and consider anything above that as log output.
+        # TODO: not sure if this code is needed/used
```

Comment:
Good catch regarding the `TODO` comment. This logic should only be required for the Docker-based Lambda executors (as lambci/docker-lambda is using stdout/stderr for different purposes). I think lines 820-832 can probably be removed.

Reply:
it's also needed for the local node runners which i modified to also output the json they return: https://github.com/localstack/localstack/pull/4267/commits/6dd15bd67ddc78a2409c05f40e23ded7ff25c7a2#diff-c01a1cd0b361abb7ab2e059723a7c44b41b32a4299f4309b61cf5616f39993e3R953

Reply:
👍 Thanks for clarifying.